src/transformers/trainer.py

```diff
@@ -1223,6 +1223,8 @@ def ipex_optimize_model(self, model, training=False, dtype=torch.float32):
             if not model.training:
                 model.train()
             model, self.optimizer = ipex.optimize(model, dtype=dtype, optimizer=self.optimizer, level="O1")
+            self.model = model
```

Comment:
No, the `self.model` attribute should not be changed, it should always keep the original model.

Reply:
Please also see this:

https://github.com/huggingface/transformers/blob/7c046c5c22850a5f1ca71381472243a739eb9d2a/src/transformers/trainer.py#L275-L290

so you can either re-use or add a new entry if you need to override the normal model as Sylvan said.

Reply:
Got it, here have removed these lines to avoid confusion. Thanks!